src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Icu.cs

```diff
@@ -364,32 +364,54 @@ private static string ConvertIcuTimeFormatString(ReadOnlySpan<char> icuFormatStr
                 char current = icuFormatString[i];
                 switch (current)
                 {
+                    case '\\':
+                        result[resultPos++] = '\\';
+                        result[resultPos++] = '\\';
+                        break;
                     case '\'':
-                        result[resultPos++] = icuFormatString[i++];
-                        while (i < icuFormatString.Length)
+                        static bool HandleQuoteLiteral(ReadOnlySpan<char> icuFormatString, ref int i, Span<char> result, ref int resultPos)
+                        {
+                            if (i + 1 < icuFormatString.Length && icuFormatString[i + 1] == '\'')
+                            {
+                                result[resultPos++] = '\\';
+                                result[resultPos++] = '\'';
+                                i++;
+                                return true;
+                            }
+                            return false;
```

Comment:
Maybe easier to have `result[resultPos++] = '\'';` before reurning false and remove this from the calling site.